kernel/kernbench.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Sudeesh John <sudeesh@linux.vnet.ibm.com>
+# Author: Harish Sriram <harish@linux.vnet.ibm.com>
+#
+# Based on code by "mbligh@google.com (Martin Bligh)"
+# https://github.com/autotest/autotest-client-tests/commits/master/kernbench
+
+import os
+import logging
+import re
+import platform
+
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import kernel
+from avocado.utils import cpu
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+log = logging.getLogger('avocado.test')
+
+
+class Kernbench(Test):
+    """
+    Kernbench compiles the kernel source fetched from the kernel.org
+    and compile it as per the configuration file. Performance figures will be
+    shown in the log file.
+    """
+
+    def time_build(self, kern,
+                   threads=None, timefile=None, make_opts=None):
+        """
+        Time the bulding of the kernel
+        """
+        os.chdir(kern.linux_dir)
+        build.make(kern.linux_dir, extra_args='clean')
+        if kern.config_path is None:
+            build.make(kern.linux_dir, extra_args='defconfig')
+        else:
+            build.make(kern.linux_dir, extra_args='oldnoconfig')
+        if make_opts:
+            build_string = "/usr/bin/time -o %s make %s -j %s vmlinux" % (
+                timefile, make_opts, threads)
+        else:
+            build_string = "/usr/bin/time -o %s make -j %s vmlinux" % (
+                timefile, threads)
+        process.system(build_string, allow_output_check='None', shell=True)
```

Comment:
i think you are interested in time taken on execution (kernel source make ) so using build.run_make is more appropriate library to use here

Reply:
`run_make` enough for only the running time of command, but `time` also calculates user, system and elapsed time.

Reply:
sure